libursa/src/encryption/mod.rs

```diff
@@ -0,0 +1,20 @@
+//! Encryption is categorized as public key encryption or symmetric encryption
+//! `symm` provides symmetric AEAD cryptographic algorithms.
+use aead::generic_array::{ArrayLength, GenericArray};
+use aead::Error;
+use rand::{rngs::OsRng, RngCore};
+
+pub mod symm;
+
+pub fn random_vec(bytes: usize) -> Result<Vec<u8>, Error> {
+    let mut value = vec![0u8; bytes];
+    let mut rng = OsRng::new().map_err(|_| Error)?;
+    rng.fill_bytes(value.as_mut_slice());
+    Ok(value)
+}
+
+pub fn random_bytes<T: ArrayLength<u8>>() -> Result<GenericArray<u8, T>, Error> {
+    Ok(GenericArray::clone_from_slice(
```

Comment:
Don't you want `GenericArray` to avoid heap allocation and allocate on stack? Using `random_vec` will lead to heap allocation due to `Vec`. Why not get the mutable slice of the `GenericArray` with ` as_mut_slice` and then pass the slice to `rng.fill_bytes`?

Reply:
GenericArray does not have a creation method to do that. Otherwise I would if you can find one let me know.